setdeviceconfig.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import yaml
+from shutil import move
+
+
+def split_yaml_params(yaml_raw, params):
+    """ 
+        returns yaml without specified params and yaml from params
+        keeping comments that are above parameters
+    """
+
+    comment_chunk = ""
+    clean_yaml = ""
+    removed_yaml = ""
+    param_regex = re.compile("^\\s*(\\S*):")
+    comment_regex = re.compile("^\\s*#")
+    for line in yaml_raw.splitlines(True):
+        if comment_regex.match(line):
+            comment_chunk += line
+        elif line.split() == "":
+            clean_yaml += comment_chunk
+            comment_chunk = ""
+        else:
+            param_match = param_regex.match(line)
+            if param_match and param_match.group(1) in params:
+                removed_yaml += comment_chunk + line
+                comment_chunk = ""
+            else:
+                clean_yaml += comment_chunk + line
+                comment_chunk = ""
+
+    return clean_yaml, removed_yaml
```

Comment:
It might be a whole lot safer and easier to use pyyaml (`import yaml`) to do this. It's already installed for Python 3 on all our pis. That'll lose comment lines but they're not all that important.

Reply:
Ye, I did that at first but then thought the comments would be nice. Unless you think this will cause future headaches. could leave it now?

Reply:
Just leave it for now